### PR TITLE
Fix gvk not found issue

### PR DESCRIPTION
**Pull request by elgnay**
This issue was introduced by https://github.com/open-cluster-management/work/pull/9

### CODE REVIEW

File: pkg/spoke/controllers/manifestwork_controller.go [hunk 1]
```diff
@@ -425,3 +405,57 @@ func buildAppliedStatusCondition(err error) workapiv1.StatusCondition {
...
+
+	if object == nil {
+		return resourceMeta, err
+	}
+
+	var gvk schema.GroupVersionKind
+	var namespace, name string
+
+	switch objectType := object.(type) {
+	case *unstructured.Unstructured:
```
Comment (deads2k) [inline pkg/spoke/controllers/manifestwork_controller.go:422]: this case doesn't collapse into the resourcehelper.GuessObjectGroupVersionKind(objectType) below?

Comment (elgnay) [reply pkg/spoke/controllers/manifestwork_controller.go:422]: I tried and found GuessObjectGroupVersionKind() works for Unstructured object. I'll refine the code in a followup pr.

**Pull request merged**